torch/csrc/jit/passes/quantization.cpp

```diff
@@ -225,6 +261,15 @@ void InsertQuantDequantNodes(std::shared_ptr<Graph>& graph) {
         blocks_to_visit.push(subblock);
       }
 
+      if (matchQParamDictKeytoObserver<std::string, float, int>
```

Comment:
This function does two separate things, which makes it more confusing than it could be:
1) It checks if the node is observer
2) If the node is observer, it finds and extracts the corresponding value from `qparam_dict`.
Could you please split it?